src/api/auth-api.ts

```diff
@@ -0,0 +1,16 @@
+import { AxiosResponse } from 'axios';
+import createHTTP from './api';
+import { SignInRequest, UserDTO, SignUpResponse } from './types';
+
+const auth = createHTTP('/auth');
+
+export default {
+  async signin(user: SignInRequest): Promise<void> {
+    await auth.post('/signin', user);
+  },
+
+  async signup(user: UserDTO): Promise<AxiosResponse<SignUpResponse>> {
+    const response = await auth.post('/signup', user);
```

Comment:
❗️
`return`-ы стоит отбивать пустой строкой для лучшей читаемости
```suggestion
    const response = await auth.post('/signup', user);
    
```